src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/CompareSymbolsCorrectlyAnalyzer.cs

```diff
@@ -215,7 +215,7 @@ static bool HasFirstTypeArgumentSymbolType(IMethodSymbol? method, INamedTypeSymb
                 collectionTypes.Contains(createdType.OriginalDefinition) &&
                 !createdType.TypeArguments.IsEmpty &&
                 IsSymbolType(createdType.TypeArguments[0], symbolType) &&
-                !objectCreation.Arguments.Any(arg => IsSymbolType(arg.Type, symbolEqualityComparerType)))
+                !objectCreation.Arguments.Any(arg => IsSymbolType(arg.Value, symbolEqualityComparerType)))
```

Comment:
@ryzngard can you please review?

Reply:
It's the second similar fix I am doing which makes me question a change on Roslyn.

Reply:
Looking at this one right after #4477

Reply:
> It's the second similar fix I am doing which makes me question a change on Roslyn.

Just to make sure I'm interpreting correctly. You hit these issues while working in Roslyn and are fixing here?

Reply:
Sorry no, I meant that this is the second issue raised on roslyn-analyzers related to getting the type on a conversion. I am wondering if the code used to work and it stopped when we have moved to a newer version of Roslyn. My point is to say, maybe there is a regression on Roslyn side regarding these operations (Type wasn't null and now is).